src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -258,9 +258,7 @@ class MarathonSchedulerActor private (
 
   def deploymentFailed(plan: DeploymentPlan, reason: Throwable): Unit = {
     logger.error(s"Deployment ${plan.id}:${plan.version} of ${plan.targetIdsString} failed", reason)
-    Future.sequence(plan.affectedRunSpecIds.map(launchQueue.purge))
-      .recover { case NonFatal(error) => logger.warn(s"Error during async purge: planId=${plan.id} for ${plan.targetIdsString}", error); Done }
-      .foreach { _ => eventBus.publish(core.event.DeploymentFailed(plan.id, plan, reason = Some(reason.getMessage()))) }
+    eventBus.publish(core.event.DeploymentFailed(plan.id, plan, reason = Some(reason.getMessage())))
```

Comment:
First actual change: we got rid of `launchQueue.purge` since 
- either deployment was canceled (and we leave _Scheduled_ instances as they were created) or i
- it was overridden by the next (forced) deployment and next `TaskReplaceActor` will take care of that